precompiles/parachain-staking/src/lib.rs

```diff
@@ -208,6 +208,27 @@ fn parse_uint256(input: &[u8]) -> Result<U256, ExitError> {
 	Ok(U256::from_big_endian(&input[0..SIZE_BYTES]))
 }
 
+/// Parses Weight Hint: u32 from a 256 bit (32 byte) slice.
+fn parse_weight_hint(input: &[u8]) -> Result<u32, ExitError> {
+	const WEIGHT_HINT_SIZE_BYTES: usize = 32;
```

Comment:
Why not  just pass the weight hints in as normal integers (etc)?

Reply:
The call data is &[u8] and we parse that into the input arguments for the runtime method. Is there any other method that already exists to parse u32 from &[u8] ? cc @JoshOrndorff

Reply:
Oh, nevermind, I didn't realize I was looking at something that was directly called from a precompile. That makes sense. :+1: